pkg/private/zip/build_zip.py

```diff
@@ -193,7 +193,14 @@ def add_tree(self, tree_top: str, destpath: str, mode: int):
         dest_dir = dest
       to_write[dest_dir] = None
       for file in files:
-        to_write[dest_dir + file] = root + '/' + file
+        if os.name == "nt":
+          # "To specify an extended-length path, use the `\\?\` prefix. For
+          # example, `\\?\D:\very long path`."[1]
+          #
+          # [1]: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+          to_write[dest_dir + file] = "\\\\?\\" + os.path.abspath(os.path.join(root, file))
+        else:
```

Comment:
Now that you are handling windows better.  I think it will be clearer if all paths go through abspath....
```
content_path = os.path.abspath(os.path.join(root, file))
if os.name == 'nt':
   to_write[...] = "\\\\?\\" + content_path
else
  to_write[...] = content_path
```
or similar

Reply:
Sure thing. I found a similar snippet in build_tar.py. I'm going to include the same fix for that in this PR, but lmk if you'd prefer it to be dealt with separately.